ZIMP_game/models/game.py

```diff
@@ -1,27 +1,189 @@
+        self.environment = Environment()
+        self.environment.shuffle_tiles()
+        self.load_tile(IndoorTile(IndoorTileName.Foyer),
+                       self.STARTING_Y, self.STARTING_X)
+        self.player = Player()  # Create a Player object
+        self.card_deck = CardDeck()  # Create a CardDeck object
+
+    def load_tile(self, tile, y_cord, x_cord):
+        """Loads tile onto game map."""
+        self.game_map[y_cord, x_cord] = tile
+        self.current_tile = [y_cord, x_cord]
+        self.current_tile_object = self.game_map[y_cord, x_cord]
+
+    def move(self, direction):
+        """Move to a new direction"""
+
+        # If moving out through the dining room to the patio
+        if ((self.current_tile_object.name == IndoorTileName.Bathroom)
+                & self.current_tile_object.white_arrows[direction]):
+            self.place_patio(direction)
+
+        # If in the Bathroom with no exits or all exits explored
+        if ((self.current_tile_object.name == IndoorTileName.Bathroom)
+                | self.is_all_exits_explored()):
+            self.break_room()
+
+        # If on a tile with no special features
+        else:
+            self.tile_direction_action(direction)
+
+    def tile_direction_action(self, direction):
+        """If there are no tiles in the next direction, draw a new tile;
+        otherwise, revisit a tile"""
+        tile_not_occupied = 0
+        if self.generate_temp_location(direction) != tile_not_occupied:
+            self.move_to_new_tile(direction)
+        else:
+            self.revisit_tile(direction)
+
+    def move_to_new_tile(self, direction):
+        """Executes when player moves to a direction with no occupied tile."""
+        if self.can_exit(direction):
+            self.draw_new_tile()
+            self.match_exit_with_entry(direction)
+            self.update_current_tile_position(direction)
+            # How to check if valid?
+
+    def can_exit(self, direction):
+        """Returns True when valid exit direction is chosen
+        for indoors/ outdoors tile"""
+        if self.current_tile_object.exits[direction]:
+            return True
+        else:
+            return False
+
+    def update_current_tile_position(self, direction):
+        """Updates current tile position based on exit direction"""
+        match direction:
+            case "north":
+                self.current_tile = [self.current_tile[0] - 1, self.current_tile[1]]
+            case "east":
+                self.current_tile = [self.current_tile[0], self.current_tile[1] + 1]
+            case "south":
+                self.current_tile = [self.current_tile[0] + 1, self.current_tile[1]]
+            case "west":
+                self.current_tile = [self.current_tile[0], self.current_tile[1] - 1]
+            case _:
+                raise ValueError("Incorrect direction name!")
+        # Add to game map
+        self.game_map[self.current_tile[0],

```

Comment:
Shouldn't it call the load_tile() method and use a random tile as the tile parameter and the new current_tile list values as the other parameters?